libyul/backends/evm/EVMCodeTransform.cpp

```diff
@@ -572,17 +576,30 @@ void CodeTransform::operator()(Block const& _block)
 	m_scope = originalScope;
 }
 
-AbstractAssembly::LabelID CodeTransform::functionEntryID(YulString _name, Scope::Function const& _function)
+void CodeTransform::createFunctionEntryID(FunctionDefinition const& _function)
 {
-	if (!m_context->functionEntryIDs.count(&_function))
-	{
-		AbstractAssembly::LabelID id =
-			m_useNamedLabelsForFunctions ?
-			m_assembly.namedLabel(_name.str(), _function.arguments.size(), _function.returns.size(), {}) :
-			m_assembly.newLabelId();
-		m_context->functionEntryIDs[&_function] = id;
-	}
-	return m_context->functionEntryIDs[&_function];
+	Scope::Function& scopeFunction = std::get<Scope::Function>(m_scope->identifiers.at(_function.name));
+	yulAssert(!m_context->functionEntryIDs.count(&scopeFunction), "");
+
+	optional<size_t> astID;
+	if (_function.debugData)
+		astID = _function.debugData->astID;
+
+	m_context->functionEntryIDs[&scopeFunction] =
+		m_useNamedLabelsForFunctions ?
+		m_assembly.namedLabel(
```

Comment:
Is there a test that showcases this change?

Reply:
Yes, it allows us to create this structure: https://github.com/ethereum/solidity/pull/11975/files/a2fb5a1b470a4bd27a20da9780fd091ff03d1a0e#diff-f45c89f87fcc7cd29d5b5122cadfea72d13562131a9bd28f9f1a946265505bf2R6